hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -1160,16 +1183,31 @@ public void setMaintenanceReplicaMinimum(int replicaCount) {
         defaultValue = "20",
         tags = { SCM, DATANODE },
         description = "A limit to restrict the total number of replication " +
-            "commands queued on a datanode. Note this is intended to be a " +
-            " temporary config until we have a more dynamic way of limiting " +
-            "load."
+            "and reconstruction commands queued on a datanode. Note this is " +
+            "intended to be a temporary config until we have a more dynamic " +
+            "way of limiting load."
     )
     private int datanodeReplicationLimit = 20;
 
     public int getDatanodeReplicationLimit() {
       return datanodeReplicationLimit;
     }
 
+    @Config(key = "datanode.reconstruction.weight",
+        type = ConfigType.INT,
+        defaultValue = "3",
+        tags = { SCM, DATANODE },
+        description = "When counting the number of replication commands on a " +
+            "datanode, the number of reconstruction commands is multiplied " +
+            "by this weight to ensure reconstruction commands use more of " +
+            "the capacity, as they are more expensive to process."
+    )
+    private int reconstructionCommandWeight = 3;
```

Comment:
I think the weight we should use might depend on the EC scheme (3-2 vs. 6-3 vs. 10-4), so I'm not sure we should have a config item for this.

Reply:
Yea this is a good point, and warrants more thought. The question is what value should be used. Eg we could use the parity number, or some functions of the nodes involved, but we may still need a config setting to "tune" it. I will create another Jira under our RM improvements epic so we can consider how best to do this.

Its difficult to pick sensible defaults for these sort of settings without testing on real hardware and data, and even then, the best setting probably depends on the hardware capabilities, network and how the data looks.